huggingface/chat-ui
# Add support for tgi multimodal models

Working for now but things that still need fixing:

- [x] Better UI for deleting an uploaded image (like a ❌ button or something)
- [x] Better UI for displaying an image in the chat
- [x] Handle resizing of images in the front-end
- [x] Add a button for uploading an image without dragging it (for mobile for example)
- [x] Disable dragging feature if `multimodal:true` is not set in the config for the model
- [x] Check image size on backend
- [x] Retry button doesn't retry with images
- [x] share conversation with images
- [x] when sending a message, show images in optimistic updates
- [x] Only allow 1 image upload max

## How to test
1. pull this pr
2. `npm run updateLocalEnv`
3. `npm run dev`

Start a conversation with IDEFICS to see the new ui

## Screenshots

![image](https://github.com/huggingface/chat-ui/assets/25119303/bbdf1a96-4543-4a38-9baf-fe5a15a0f812)
![image](https://github.com/huggingface/chat-ui/assets/25119303/b4335585-e08c-491c-9c2b-c05298b52023)
